eventlet/semaphore.py

```diff
@@ -78,9 +78,18 @@ def acquire(self, blocking=True, timeout=None):
         When invoked with blocking set to false, do not block. If a call without
         an argument would block, return false immediately; otherwise, do the
         same thing as when called without arguments, and return true.
+
+        Timeout value must be strictly positive.
         """
-        if not blocking and timeout is not None:
-            raise ValueError("can't specify timeout for non-blocking acquire")
+        if timeout == -1:
+            timeout = None
+        if timeout is not None and timeout < 0:
+            raise ValueError("timeout value must be strictly positive")
```

Comment:
> Exception says "strictly positive" ( x > 0 ), while code allows timeout=0 just fine.

Hum... yes... I copied code from the C implementation of threading.Lock, to try to be as close as possible to threading.Lock.

Example:

```
$ python3
Python 3.4.1 (default, Nov  3 2014, 14:38:10) 
>>> import threading
>>> l=threading.Lock()
>>> l.acquire(timeout=-2)
...
ValueError: timeout value must be strictly positive
>>> l.acquire(timeout=0)
True
>>> l.acquire(timeout=0)
False
```